### ISSUE TITLE
This samples have a problerm

### ISSUE THREAD

**Issue by czh249899518** [spring-projects-experimental/spring-authorization-server#165]
I've start three services

Go to http://localhost:8080/index, click **Authorization Code**  .

Redirect to http://localhost:9000/login,  do login by sample user ,login in (http://localhost:9000/oauth2/authorize?response_type=code&client_id=messaging-client&scope=message.read%20message.write&state=4kZek4lBZpehCTLS9l-1itLphcwAwKYfzexq0z7LmAI%3D&redirect_uri=http://localhost:8080/authorized)

Click [Submit consent] redirect to (http://localhost:8080/authorized?code=V-9R_2M-O_abn2Gw0Owrzvno_6HqCwXHsEzWchb7UgXQVZs_0WKaN2M-45fowB6aL18TwMHQIfTLd-qvVTHkZUFs-7d6YVuqRA8TlIlrIXDV_GpsfT-qo5Nh0O3GWJMW&state=4kZek4lBZpehCTLS9l-1itLphcwAwKYfzexq0z7LmAI%3D)

A 400 error occurred

I think the samples [client-AuthorizationController] is no such method(/authorized,params=code)

**Comment by jgrandja** [comment]
@czh249899518 I just tried the sample on latest `master` and everything is working as expected. Are you using a browser or simulating with Postman / curl?

Please try again using latest master and browser.

**Comment by czh249899518** [comment]
> @czh249899518 I just tried the sample on latest `master` and everything is working as expected. Are you using a browser or simulating with Postman / curl?
> 
> Please try again using latest `master` and a browser.


I use Chrome browser to test.
Is it because my redirect-Url hasn't changed?
**redirect-uri: "{baseUrl}/authorized"** from [client] application.yml

**Do you have email? I'll record a video for you to see**

[client] have warn :WARN 8620 --- [nio-8080-exec-6] .w.s.m.s.DefaultHandlerExceptionResolver : Resolved [org.springframework.web.bind.UnsatisfiedServletRequestParameterException: Parameter conditions "error" not met for actual request parameters: code={hmKC9ctIoK_WS0wmn0j4VSVmtkq4LWhZrxBMTza4kPSyVe2LizU7lx5J-lDF-S1pS1vtdTE4ywPzqaRu8XixY_zYzYdWAXvfg55TSA071jPrMj7O2RVBSilwTeO9MkU-}, state={FsgjJdzfNn44owEspJDt35TiEnv0_GXECCskClOEHTM=}]

**Comment by jgrandja** [comment]
@czh249899518 

> Is it because my redirect-Url hasn't changed?

`redirect-uri: "{baseUrl}/authorized"` should not change as it's registered the same in the [Authorization Server](https://github.com/spring-projects-experimental/spring-authorization-server/blob/d97235d0bbf660f81d5399a2a1c52b92f782ace3/samples/boot/oauth2-integration/authorizationserver/src/main/java/sample/config/AuthorizationServerConfig.java#L49).

For your reference, please review [Redirection Endpoint](https://tools.ietf.org/html/rfc6749#section-3.1.2) and [Authorization Response](https://tools.ietf.org/html/rfc6749#section-4.1.2).

> I think the samples [client-AuthorizationController] is no such method(/authorized,params=code)

There is no declared `@Controller` method for `/authorized` and it's not needed in the application. This is only required for the `oauth2-client` to receive the `code` on the Authorization Response callback, which it then exchanges the code for an access token.

As far as sending a video, you can attach it here or post it somewhere and provide the link.

**Comment by jgrandja** [comment]
@czh249899518 What would be best is that you put together a [minimal sample](https://stackoverflow.com/help/minimal-reproducible-example) that reproduces the issue so I can see exactly what the issue is.

**Comment by czh249899518** [comment]
Is it right to return error 400&nbsp; ?
I think should be return [messages1,2,3]&nbsp; after authorize


There was an unexpected error (type=Bad Request, status=400).


Because can't send vedio，please read from BaiduCloudDisk


Url: https://pan.baidu.com/s/1n3fxyat6y3xfIuLEQc5P-g&nbsp;
password: 6y97&nbsp;


if you can't open it，I'll change other way






------------------ 原始邮件 ------------------
发件人:                                                                                                                        "spring-projects-experimental/spring-authorization-server"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年11月26日(星期四) 晚上8:56
收件人:&nbsp;"spring-projects-experimental/spring-authorization-server"<spring-authorization-server@noreply.github.com&gt;;
抄送:&nbsp;"徽💯"<249899518@qq.com&gt;;"Mention"<mention@noreply.github.com&gt;;
主题:&nbsp;Re: [spring-projects-experimental/spring-authorization-server] This samples have a problerm (#165)





 
@czh249899518 What would be best is that you put together a minimal sample that reproduces the issue so I can see exactly what the issue is.

**Comment by jgrandja** [comment]
@czh249899518 As mentioned in last comment

> It would be best if you put together a [minimal sample](https://stackoverflow.com/help/minimal-reproducible-example) that reproduces the issue so I can see exactly what is going on.

The video won't help me get to the root of the issue.

Until I get a sample that reproduces the issue, it will be very difficult to help resolve.

**Comment by czh249899518** [comment]
I think your example is not complete.

My expectation is that after clicking the authorization button, it will redirect to the home page and display message 1, message 2, and message 3.

But redirect 400 error page.

I'm  using the latest version of the master branch.

If that's what this example expects,Then you can ignore my issue

**Comment by jgrandja** [comment]
@czh249899518 The sample is working as expected. The messages (1, 2 ,3) are being displayed after the authorization flow completes. I haven't seen a 400 error as of yet and no one else has reported this issue. The sample has been around for a while now.

There is something going on with your setup / environment.

The only way for me to help you is to reproduce on my end.

> It would be best if you put together a [minimal sample](https://stackoverflow.com/help/minimal-reproducible-example) that reproduces the issue so I can see exactly what is going on.

**Comment by czh249899518** [comment]
I have found the problem.


because I changed the [client-application.yml]&nbsp;auth-server to localhost.  


Because my negligence caused you trouble,I'm so sorry


Thank you 








 
@czh249899518 The sample is working as expected. The messages (1, 2 ,3) are being displayed after the authorization flow completes. I haven't seen a 400 error as of yet and no one else has reported this issue. The sample has been around for a while now.
 
There is something going on with your setup / environment.
 
The only way for me to help you is to reproduce on my end.
  
It would be best if you put together a minimal sample that reproduces the issue so I can see exactly what is going on.
